tutorials/dataframe/tdf010_trivialDataSource.C

```diff
@@ -0,0 +1,47 @@
+/// \file
+/// \ingroup tutorial_tdataframe
+/// \notebook
+/// This tutorial illustrates how use the TDataFrame in combination with a
+/// TDataSource. In this case we use a TTrivialDS, which is nothing more
+/// than a simple generator: it does not interface to any existing dataset.
+/// The TTrivialDS has a single column, col0, which has value n for entry n.
+///
+/// \macro_code
+///
+/// \date September 2017
+/// \author Danilo Piparo
+
+using TDS = ROOT::Experimental::TDF::TDataSource;
+using TTrivialDS = ROOT::Experimental::TDF::TTrivialDS;
+
+int tdf010_trivialDataSource()
+{
+   auto nEvents = 128U;
+
+   std::unique_ptr<TDS> tds(new TTrivialDS(nEvents));
+   ROOT::Experimental::TDataFrame d_s(std::move(tds));
```

Comment:
should we use the factory function here too?

Reply:
not sure. Why not.